### PR TITLE
Adding Secrets to Job and Build (fixing bug with falsey values)

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This adds secrets to job and build model.  As well as fixing a bug when updating values to `false`

### CODE REVIEW

**Comment by petey** [comment]
👍

File: lib/baseFactory.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ class BaseFactory {
 
         // Filter for valid keys
         this.model.allKeys.forEach((key) => {
-            if (config[key]) {
+            if (config[key] !== undefined) {
```
Comment (d2lam) [inline lib/baseFactory.js:66]: Good catch

File: lib/job.js [hunk 2]
```diff
@@ -43,6 +44,38 @@ class Job extends BaseModel {
...
 
     /**
+     * Lazy load the list of secrets
+     * @property secrets
+     * @return {Promise}    Resolves to a list of secrets
+     */
+    get secrets() {
+        delete this.secrets;
+
+        const secretNames = hoek.reach(this.permutations, '0.secrets', { default: [] });
```
Comment (d2lam) [inline lib/job.js:54]: what is `0.secrets`?

Comment (d2lam) [inline lib/job.js:54]: Oh I see. It's looking for `this.permutations[0].secrets`
But why? Is it always going to be index 0?

Comment (stjohnjohnson) [reply lib/job.js:54]: reads the first item in the array and the field secrets

**Comment by d2lam** [comment]
Just some questions, the rest LGTM 👍

File: lib/job.js (hunk 2 above)
Comment (stjohnjohnson) [reply lib/job.js:54]: Ahh, that's because secrets cannot be mutated right now, so all permutations will have the same list of secrets.

Comment (d2lam) [reply lib/job.js:54]: Oh I see. Makes sense 👍

**Pull request merged**